pkg/ui/workspaces/cluster-ui/src/databaseTablePage/databaseTablePage.tsx

```diff
@@ -304,159 +314,74 @@ export class DatabaseTablePage extends React.Component<
-    if (lastReset.isSame(this.minDate)) {
-      return <>Last reset: Never</>;
-    } else {
-      return (
-        <>
-          Last reset: <Timestamp time={lastReset} format={DATE_FORMAT_24_TZ} />
-        </>
-      );
-    }
-  }
-
-  private getLastUsed(indexStat: IndexStat) {
-    // This case only occurs when we have no reads, resets, or creation time on
-    // the index.
-    if (indexStat.lastUsed.isSame(this.minDate)) {
-      return <>Never</>;
-    }
-    return (
-      <>
-        Last {indexStat.lastUsedType}:{" "}
-        <Timestamp time={indexStat.lastUsed} format={DATE_FORMAT} />
-      </>
-    );
-  }
-
-  private renderIndexRecommendations = (
-    indexStat: IndexStat,
-  ): React.ReactNode => {
-    const classname =
-      indexStat.indexRecommendations.length > 0
-        ? "index-recommendations-icon__exist"
-        : "index-recommendations-icon__none";
-
-    if (indexStat.indexRecommendations.length === 0) {
-      return (
-        <div>
-          <CircleFilled className={cx(classname)} />
-          <span>None</span>
-        </div>
-      );
-    }
-    return indexStat.indexRecommendations.map((recommendation, key) => {
-      let text: string;
-      switch (recommendation.type) {
-        case "DROP_UNUSED":
-          text = "Drop unused index";
-      }
-      return (
-        <Tooltip
-          key={key}
-          placement="bottom"
-          title={
-            <div className={cx("index-recommendations-text__tooltip-anchor")}>
-              {recommendation.reason}{" "}
-              <Anchor href={performanceTuningRecipes} target="_blank">
-                Learn more
-              </Anchor>
-            </div>
-          }
-        >
-          <CircleFilled className={cx(classname)} />
-          <span className={cx("index-recommendations-text__border")}>
-            {text}
-          </span>
-        </Tooltip>
-      );
-    });
-  };
-
-  private renderActionCell = (indexStat: IndexStat): React.ReactNode => {
-    const isCockroachCloud = useContext(CockroachCloudContext);
-    if (isCockroachCloud || indexStat.indexRecommendations.length === 0) {
-      return <></>;
-    }
-
-    const query = indexStat.indexRecommendations.map(recommendation => {
-      switch (recommendation.type) {
-        case "DROP_UNUSED":
-          return `DROP INDEX ${QuoteIdentifier(
-            this.props.name,
-          )}@${QuoteIdentifier(indexStat.indexName)};`;
+  private indexStatsColumns(): ColumnDescriptor<IndexStat>[] {
+    const indexStatsColumns: ColumnDescriptor<IndexStat>[] = [
+      {
+        name: "indexes",
+        title: "Indexes",
+        hideTitleUnderline: true,
+        className: cx("index-stats-table__col-indexes"),
+        cell: indexStat => (
+          <NameCell
+            indexStat={indexStat}
+            tableName={this.props.name}
+            showIndexRecommendations={this.props.showIndexRecommendations}
+            isCockroachCloud={this.context}
```

Comment:
Where does this `this.context` come from? I've never seen this pattern before so just curious about it!

Reply:
This is the pattern I came across in the old React [docs](https://legacy.reactjs.org/docs/context.html#classcontexttype), describing how to consume context in a class component: